File: programmers/LV1/가장 가까운 같은 글자.js
// 가장 가까운 같은 글자
function solution(s) {
  let box = [];
  let answer = [];

  [...s].forEach((str) => {
    if (!box.includes(str)) {
      answer.push(-1);
    } else {
      answer.push(box.length - box.lastIndexOf(str));
    }

    box.push(str);
  });

  return answer;
}

File: programmers/python/가장 가까운 같은 글자.py
# 가장 가까운 같은 글자
def solution(s):
    answer = []
    box = dict()
    
    for i in range(len(s)):
        if s[i] not in box:
            answer.append(-1)
        else:
            answer.append(i - box[s[i]])
        box[s[i]] = i
        
    return answer